apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -577,12 +576,13 @@ impl FetchDependencyGraphNodePath {
                 None => self.possible_types.clone(),
                 Some(tcp) => {
                     let element_possible_types = self.schema.possible_runtime_types(tcp.clone())?;
-                    element_possible_types
```

Comment:
Bugs:
- In the JS code, we avoid having to re-sort because we remove elements from `this.possibleTypes` instead of from the type condition's possible runtime types. Otherwise, we have to sort. I've updated the Rust code to similar remove from `self.possible_types`, to avoid the sort.